### DISCUSSION TITLE
Can I install packages manually?

### DISCUSSION THREAD

**Discussion by GuanWX** [mason-org/mason.nvim#659] [Q&A] [answered]
Can I install packages manually?

**Comment by yutkat** [comment]
`:MasonInstall <package>`

**Comment by GuanWX** [comment]
I mean, install through a tar package. This is because my network does not support direct installation.

**Comment by GuanWX** [comment, marked as answer]
I know how to installing package manually!

- 1.  Creating a directory in `/.local/share/nvim/mason/packages`, the directory's name is the package's name you want to install;
- 2.  Decompress the tar package
- 3.  Create a soft link in `/.local/share/nvim/mason/bin`, link to  `../packages/xxx/bin/xxxxxx`

**Reply by KarateSnowMachine** [reply]
This is less than ideal since you can't really automate this, but thank you for your answer, it helped me get my stuff running.
